script.js

```diff
@@ -1,12 +1,377 @@
+  rootElem.appendChild(ourSection);
+
+  for (let episodeIndex = 0; episodeIndex < episodesList.length; episodeIndex++) {
+    let episodeContent = document.createElement("article");
+    episodeContent.classList.add("wrapper");
+
+    let episodeTitle = document.createElement("h3");
+    episodeTitle.classList.add("episodeName");
+
+    let imageContainer = document.createElement("div");
+    imageContainer.classList.add("EpisodePictureContainer")
+
+    let episodeImage = document.createElement("img");
+    episodeImage.classList.add("episodePicture")
+    imageContainer.appendChild(episodeImage)
+
+    let episodeSummary = document.createElement("div");
+    episodeSummary.classList.add("episodeDescription")
+
+    episodeContent.append(episodeTitle, imageContainer, episodeSummary);
+    ourSection.append(episodeContent);
+
+    addContentToEachEpisode(episodesList, episodeIndex);
+  }
+}
+
+function addContentToEachEpisode(list, index) {
+  let title = document.getElementsByClassName("episodeName")[index];
+  title.innerHTML = `${list[index].name} - ${getSeasonAndEpisodeNumber(list, index)}`;
+
+  let picture = document.getElementsByClassName("episodePicture")[index];
+  picture.setAttribute("src", `${getImage(list, index)}`)
+
+  let description = document.getElementsByClassName("episodeDescription")[index];
+  description.innerHTML = `${list[index].summary}`;
+}
+
+function getImage(list, index) {
+  if (list[index].image === null) {
+    return "https://upload.wikimedia.org/wikipedia/commons/a/ac/No_image_available.svg";
+  } else {
+    return list[index].image.medium;
+  }
+
+}
+
+function getSeasonAndEpisodeNumber(list, index) {
+  const pattern = 10;
+
+  let seasonNumber = `S${(list[index].season < pattern ? "0" : "") + list[index].season}`
+  let episodeNumber = `E${(list[index].number < pattern ? "0" : "") + list[index].number}`
+
+  return `${seasonNumber + episodeNumber}`
+}
+
+//level 200
+
+function addSearchBar() {
+  let navigationWrapper = document.querySelector(".navigationWrapper")
+
+  let pageNavigation = document.createElement("nav");
+  navigationWrapper.appendChild(pageNavigation);
+
+  let searchBar = document.createElement("input");
+  searchBar.setAttribute("type", "text");
+  searchBar.setAttribute("placeholder", "Search through the page");
+  searchBar.setAttribute("id", "mySearchBar");
+  pageNavigation.appendChild(searchBar)
+
+  let searchResult = document.createElement("label")
+  searchResult.classList.add("mySearchResult")
+
+  let getResultsOnPage = document.getElementsByClassName("wrapper");
+  searchResult.innerText = `Displaying ${getResultsOnPage.length} results `;
+  pageNavigation.appendChild(searchResult)
+
+  searchBar.addEventListener("keyup", searchThroughThePage)
+}
+
+function searchThroughThePage(e) {
+  let searchString = e.target.value.toUpperCase();
+
+  let getResultsOnPage = document.getElementsByClassName("wrapper")
+  let searchResult = document.getElementsByClassName("mySearchResult")
```

Comment:
I can see a bug on your page. After I do a search, the label with the number of displayed results isn't updating. The reason why is in this line (line 129).

Using your debugging skills, can you work out why line 129 is causing line 142 to not work.

Reply:
@louisechow you are right, I get an HTMLObjectCollection when using `getElementsByClassName` so I used `querySelector` , but what is confusing me now, is how come it worked when I was at level 400